pkgs/development/compilers/llvm/12/default.nix

```diff
@@ -181,13 +182,28 @@ let
       '' + mkExtraBuildCommands0 cc;
     };
 
+    lldClangNoCompilerRtWithLibc = wrapCCWith rec {
+      cc = tools.clang-unwrapped;
+      libcxx = null;
+      bintools = wrapBintoolsWith {
+        inherit (tools) bintools;
+      };
+      extraPackages = [ ];
+      extraBuildCommands = mkExtraBuildCommands0 cc;
+    };
+
   });
 
   libraries = lib.makeExtensible (libraries: let
     callPackage = newScope (libraries // buildLlvmTools // { inherit stdenv cmake libxml2 python3 isl release_version version fetch; });
   in {
 
     compiler-rt = callPackage ./compiler-rt ({ inherit llvm_meta; } //
+      (lib.optionalAttrs (stdenv.hostPlatform.useLLVM or false) {
+        stdenv = overrideCC stdenv buildLlvmTools.lldClangNoCompilerRtWithLibc;
+      }));
+
+    compiler-rt-nolibc = callPackage ./compiler-rt ({ inherit llvm_meta; } //
```

Comment:
I would have `compiler-rt` and `compiler-rt-libc`.

Reply:
Ok.